### PR TITLE
Repeat Record Couch to SQL fixes

### PR DESCRIPTION
**Pull request by millerdev**
For bugs encountered while migrating data. Also adds a new `--debug` option to the migration command to more easily get detailed information about migration errors.

:fish: Review by commit.

### Safety story

These changes only affect the management command that is being used to migrate repeat records from Couch to SQL.

### Automated test coverage

Some.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py [hunk 1]
```diff
@@ -419,6 +425,11 @@ def iter_items(docs):
                 traceback.print_exc(file=logfile)
                 aborted = True
                 print("Aborted.")
+            except Exception as exc:
+                if not debug:
+                    raise
+                traceback.print_exception(type(exc), exc, exc.__traceback__)
+                pdb.post_mortem(exc.__traceback__)
```
Comment (gherceg) [inline corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py:432]: very cool

File: corehq/motech/repeaters/tests/test_couchsqlmigration.py [hunk 2]
```diff
@@ -461,6 +461,16 @@ def test_migrate_record_with_no_attempts(self):
         obj = SQLRepeatRecord.objects.get(couch_id=doc_id)
         self.assertFalse(obj.attempts)
 
+    def test_migrate_record_erroneous_next_check(self):
+        doc, _ = self.create_repeat_record()
+        doc.update(succeeded=True, next_check=datetime.utcnow().isoformat() + 'Z')
+        doc_id = self.db.save_doc(doc)["id"]
+        with templog() as log, patch.object(transaction, "atomic", atomic_check):
+            call_command('populate_repeatrecords', log_path=log.path)
+            self.assertNotIn('has differences:', log.content)
```
Comment (gherceg) [inline corehq/motech/repeaters/tests/test_couchsqlmigration.py:470]: I'm new to this code. What exactly is the need for this assertion in these tests? It seems like the other assert is what is really testing the intention of this test (that a record is successfully migrated with next_check set to None if we didn't like the format).

File: corehq/motech/repeaters/tests/test_couchsqlmigration.py [hunk 3]
```diff
@@ -461,6 +461,16 @@ def test_migrate_record_with_no_attempts(self):
         obj = SQLRepeatRecord.objects.get(couch_id=doc_id)
         self.assertFalse(obj.attempts)
 
+    def test_migrate_record_erroneous_next_check(self):
+        doc, _ = self.create_repeat_record()
+        doc.update(succeeded=True, next_check=datetime.utcnow().isoformat() + 'Z')
```
Comment (gherceg) [inline corehq/motech/repeaters/tests/test_couchsqlmigration.py:466]: The outcome would be the same for a repeat record that succeeded or was cancelled. What is the expected behavior if a repeat record was in a different state with an erroneous next check value? Or do you not expect to encounter that situation? Should a hard failure be documented in tests as well?

**Review by gherceg — COMMENTED** [review]
Just some small comments about the test.

File: corehq/motech/repeaters/tests/test_couchsqlmigration.py (hunk 2 above)
Comment (millerdev) [reply corehq/motech/repeaters/tests/test_couchsqlmigration.py:470]: This assertion simply verifies that the migration did not result in unexpected differences. You are correct that the other assert is the most important.

File: corehq/motech/repeaters/tests/test_couchsqlmigration.py (hunk 3 above)
Comment (millerdev) [reply corehq/motech/repeaters/tests/test_couchsqlmigration.py:466]: So far I have only encountered Couch documents that have `succeeded == True` and a non-null `next_check` value. If the state was _pending_ or _failed_ and `next_check` was null then that would also be a problem, but I have not encountered that (yet).

> Should a hard failure be documented in tests as well?

Not quite following. The database constraint on `next_check` will cause a hard failure if the record state is not compatible with the value of `next_check`. That is covered by various tests, although not tested explicitly. The check constraint is declarative and I don't think we need to add more tests around it

https://github.com/dimagi/commcare-hq/blob/17cfe64fc09b82e86d19870bfc3d6f8bb6cd8735/corehq/motech/repeaters/models.py#L1369-L1374

Comment (gherceg) [reply corehq/motech/repeaters/tests/test_couchsqlmigration.py:466]: Wasn't aware of the constraint. Thanks for the thorough response!

**Review by gherceg — APPROVED** [review]

**Pull request merged**